src/contracts/OldERC721Wrapper.ts

```diff
@@ -0,0 +1,50 @@
+import { Address } from 'viem';
+
+import { Wallet } from '@/contracts';
+import { BaseContract } from '@/contracts/BaseContract';
+import { oldErc721WrapperAbi } from '@/generated/blockchain/oldERC721Wrapper';
+
+export class OldERC721Wrapper extends BaseContract<typeof oldErc721WrapperAbi> {
+  constructor({
+    walletClient,
+    contractAddress,
+  }: {
+    walletClient: Wallet;
+    contractAddress: Address;
+  }) {
+    super({
+      walletClient,
+      address: contractAddress,
+      abi: oldErc721WrapperAbi,
+    });
+  }
+
+  async wrapOldERC721({ tokenId }: { tokenId: bigint }) {
+    const txMint = await this.safeContractWrite.wrap([tokenId]);
+
+    const waitMined = async () => {
+      const receipt = await this.bcClient.waitForTransactionReceipt({ hash: txMint });
+      return receipt;
+    };
+
+    return {
+      txHash: txMint,
+      waitMined,
+    };
+  }
+
+  async unwrap(tokenId: bigint) {
+    const tx = await this.safeContractWrite.unwrap([tokenId]);
+    const waitMined = async () => {
+      const receipt = await this.bcClient.waitForTransactionReceipt({
+        hash: tx,
+      });
+      return receipt;
+    };
+
+    return {
+      txHash: tx,
+      waitMined,
```

Comment:
You can delete the waitMined definition and do:

waitMined: () => this.bcClient.waitForTransactionReceipt({
        hash: tx,
      })